crm_compassion/models/interaction_resume.py

```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
-                    AND config.name != 'Default communication'
-                WHERE mail.state = ANY (ARRAY ['sent', 'received'])
-                AND (p.contact_id = %s OR p.id = %s)
-                )
-    -- incoming messages from partners
-            UNION (
-              SELECT
-                'Email' as communication_type,
-                m.date as communication_date,
-                COALESCE(p.contact_id, p.id) AS partner_id,
-                m.email_from as email,
-                m.subject as subject,
-                REGEXP_REPLACE(m.body, '<img[^>]*>', '') AS body,
-                'in' AS direction,
-                0 as phone_id,
-                0 as email_id,
-                m.id as message_id,
-                0 as paper_id,
-                NULL as tracking_status
-                FROM "mail_message" as m
-                JOIN res_partner p ON m.author_id = p.id
-                WHERE m.subject IS NOT NULL
-                AND m.message_type = 'email'
-                AND (p.contact_id = %s OR p.id = %s)
-                )
-                """, [partner_id] * 8)
-        for row in self.env.cr.dictfetchall():
-            # ensure that "Example <ex@exmaple.com>" is correctly printed
-            row['email'] = html_sanitize(row['email'])
-            self.create(row)
+        original_partner = self.env['res.partner'].search([
+            ('id', '=', partner_id)
+        ])
+        email_address = original_partner.email
+        partners_with_same_email_ids = self.env['res.partner'].search([
+            ('email', '=', email_address)
+        ]).mapped("id")
+
+        for partner_id in partners_with_same_email_ids:
+            self.search([('partner_id', '=', partner_id)]).unlink()
+            self.env.cr.execute("""
+                    SELECT
+                        'Paper' as communication_type,
+                        pcj.sent_date as communication_date,
+                        COALESCE(p.contact_id, pcj.partner_id) AS partner_id,
+                        NULL AS email,
+                        COALESCE(c.name, pcj.subject) AS subject,
+                        REGEXP_REPLACE(pcj.body_html, '<img[^>]*>', '') AS body,
+                        'out' AS direction,
+                        0 as phone_id,
+                        0 as email_id,
+                        0 as message_id,
+                        pcj.id as paper_id,
+                        NULL as tracking_status
+                        FROM "partner_communication_job" as pcj
+                        JOIN res_partner p ON pcj.partner_id = p.id
+                        FULL OUTER JOIN partner_communication_config c
+                            ON pcj.config_id = c.id
+                            AND c.name != 'Default communication'
+                        WHERE pcj.state = 'done'
+                        AND pcj.send_mode = 'physical'
+                        AND (p.contact_id in ({}) OR p.id in ({}))
```

Comment:
==> These parameters should be replaced by `original_partner_id`. We don't want to mix the other partners for physical communications.
```suggestion
                        AND (p.contact_id = %s OR p.id = %s)
```